### DISCUSSION TITLE
What happens in case the network fails during a client request?

### DISCUSSION THREAD

**Discussion by sunilguptasg** [supabase/supabase#14555] [Questions] [answered]
I am using Supabase with Flutter and need to manage a few error cases:

1> select: in this case I check the network connectivity status before making the call, and in case a ClientException is thrown, I wait for the network state to be back and retry.

2> insert/update: I check the network connectivity before making the call - but am unsure what to do in case of a ClientException with the insert/update. Did the insert/update happen? How will I know? Does Supabase rollback the insert/update in case the network fails before the response is delivered?

What should I be doing to manage network issues with Supabase and Flutter?

**Comment by souvik666** [comment]
Hey @sunilguptasg  it might be related to this  https://github.com/supabase/supabase/issues/472

**Comment by sunilguptasg** [comment]
No, this is not related to database errors that need a rollback. The question pertains to network issues between the client and the Supabase server. 

I make an insert request - the database receives it. Before the client gets the response back, the network fails (and in Flutter causes a ClientException). 

My question is - did the insert complete? Or is it guaranteed that if the response sent to client fails, the specific insert will fail?

=========

On a secondary note - even if I move my business logic to functions, how can the client know if the insert has completed on getting a ClientException?

**Comment by souvik666** [comment]
sorry i got the question wrong

**Comment by GaryAustin1** [comment, marked as answer]
https://github.com/PostgREST/postgrest/issues/1424

Seems your question.  And answer is you need to make sure retries don't create new data.  Don't use incrementing primary key or at least have a unique column with value(UUID?) from the client.
